_posts/2021-09-23-reassign-partitions.md

```diff
@@ -0,0 +1,282 @@
+
+## Why use the Partition Reassignment tool?
+
+The Partition Reassignment tool can help you address a variety of use cases.
+
+Some of these are listed here :-
+
+1. There might be cases when you want to shutdown/remove your old broker pod from the cluster and start with some new broker pod.
+   In this scenario, the new pod won't get the partitions assigned to it. So you will have to reassign the
+   partitions among all the brokers.
+
+2. It can also be used at times when you want to expand you cluster.
+   You can add more brokers to your clusters and after that use the partition reassignment tool to assign the
+   partitions to the newly generated brokers.
+
+## Example time
+
+Let's have a look at an example which will show us how we can use the partition reassignment tool.
+In this example we will be taking a look at how the `--generate` mode of the partition reassignment tool is used.
+We will generate the JSON data which will be used in the `reassignment.json` file.
+The Kafka Cluster that we will use will be configured to use TLS encryption, and we will be using simple authentication.
+The configurations can be changed accordingly. Strimzi provides us with many authentication types like *TLS*, *SCRAM-SHA-512*, *OAUTH* and *PLAIN*.
+
+### Preparation
+
+This example will use a Kafka cluster deployed with the Strimzi Cluster Operator.
+Before we start with anything, we have to install the Strimzi Cluster Operator and deploy the Kafka cluster.
+You can install the Cluster Operator with any installation method you prefer.
+And then deploy the Kafka cluster with the TLS client authentication enabled on port 9093.
+
+Example Kafka configuration for TLS authentication
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    replicas: 3
+    listeners:
+      - name: tls
+        port: 9093
+        type: internal
+        tls: true
+        authentication:
+          type: tls
+    authorization:
+      type: simple
+    config:
+      offsets.topic.replication.factor: 3
+      transaction.state.log.replication.factor: 3
+      transaction.state.log.min.isr: 2
+    storage:
+      type: jbod
+      volumes:
+      - id: 0
+        type: persistent-claim
+        size: 100Gi
+        deleteClaim: true
+  zookeeper:
+    replicas: 3
+    storage:
+      type: persistent-claim
+      size: 100Gi
+      deleteClaim: true
+  entityOperator:
+    topicOperator: {}
+    userOperator: {}
+```
+
+Once the cluster is running, lets deploy some topics where we will send and receive the messages.
+Here is an example topic configuration. You can change it to suit your requirements.
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaTopic
+metadata:
+  name: my-topic
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  partitions: 10
+  replicas: 3
+  config:
+    retention.ms: 7200000
+    segment.bytes: 1073741824
+    # ...
+```
+
+Once the cluster is running, we will need a user.
```

Comment:
You have `Once the cluster is running` twice ... here and before the previous example. I guess once would be enough.